components/server/src/ome/services/ThumbnailBean.java

```diff
@@ -994,7 +994,7 @@ public void createThumbnailsByLongestSideSet(Integer size,
                 pixelsId = pixels.getId();
                 settings = ctx.getSettings(pixelsId);
                 thumbnailMetadata = ctx.getMetadata(pixelsId);
-                if (!PROGRESS_VERSION.equals(thumbnailMetadata.getVersion())) {
+                if (inProgress && !PROGRESS_VERSION.equals(thumbnailMetadata.getVersion())) {
```

Comment:
@jburel and I were looking at what is exactly causing dirty thumbnail. Looks like inProgress was missing here. Any objections to that single fix? I will rebase this against develop and see if combining with https://github.com/openmicroscopy/openmicroscopy/pull/5073 would resolve the issue